core/quantity/src/Parser.ts

```diff
@@ -786,11 +787,18 @@ export class Parser {
     if(this.isParseError(parsedResult)) {
       return parsedResult;
     }
-
-    let magnitude = parsedResult.value;
     const revolution = this.getRevolution(spec);
-    magnitude = this.normalizeAngle(magnitude, revolution);
     const quarterRevolution = revolution / 4;
+
+    if (parsedResult.value > quarterRevolution) {
+      return { ok: false, error: ParseError.BearingAngleOutOfRange };
+    }
+
+    let magnitude = parsedResult.value;
+    if (magnitude < -quarterRevolution || magnitude > quarterRevolution) {
```

Comment:
couldn't you have the left side of the OR in line 793? The RHS seems like it's already checked in line 793 as well